### PR TITLE
introduce [OFFLINENUM]

### LINKED ISSUE #987: Offline found count
Suggestion from the german geoclub forum:
For offline logging it would be nice to have a found count as well. So I suggest to introduce a new variable [OFFLINENUM] that relies on some start value that you manually set in the settings and is automatically inceased with every log. It lays the responsibility o n the user that it is correctly set, but allows for convinient offline caching and logging.

### PR DESCRIPTION
**Pull request by fm-sys**
fix #987 
fix https://github.com/cgeo/cgeo/pull/8603#issuecomment-668718020

It uses the `DBExtension` to store the found number for each geocaching service. The values will be updated each time the main activity is called and a connection to a geocaching service is established.

when the signature variable is used, all offlinelogs of the specific connector will be summed up added to the cached found number

For additional description see https://github.com/cgeo/cgeo/issues/987#issuecomment-669544510

### CODE REVIEW

File: main/src/cgeo/geocaching/storage/extension/FoundNumCounter.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        final DataStore.DBExtension temp = load(type, serviceName);
+        return null == temp ? null : new FoundNumCounter(temp);
+    }
+
+    public static void updateFoundNum(final String serviceName, final int foundNum) {
+
+        //avoid recreation of same key/value pair
+        final FoundNumCounter f = load(serviceName);
+        if (f != null) {
+            if (f.getCounter(false) == foundNum) {
```
[the remark this answers is not in the document]
Comment (fm-sys) [reply main/src/cgeo/geocaching/storage/extension/FoundNumCounter.java:33]: there's nothing I can do about that here... If I combine these if statements there will be the error `"null object has no method getCounter"` or something like that...

[the remark this answers is not in the document]
Comment (moving-bits) [reply main/src/cgeo/geocaching/storage/extension/FoundNumCounter.java:33]: Why is that? Normally evaluation for a combined boolean expression stops as soon as the result is fixed, so if you write `f != null && f.getCounter(false) == foundNum)` it should not evaluate the second term if `f == null`.

[the remark this answers is not in the document]
Comment (fm-sys) [reply main/src/cgeo/geocaching/storage/extension/FoundNumCounter.java:33]: OK, I will give it another try...

**Comment by fm-sys** [comment]
Fixes:
- update the cached found number also after submitting a log
- also handle WEBCAM_PHOTO_TAKEN and ATTENDED for counting +1
- change working mode of [NUMBER] instead of introducing a new variable
(all these changes are discussed in #987)

PR is now ready to review from my side...

**Comment by fm-sys** [comment]
retest this please

**Comment by fm-sys** [comment]
retest this please

**Comment by fm-sys** [comment]
retest this please

**Comment by moving-bits** [comment]
@fm-sys Thanks for your contribution!

**Pull request merged**